python_selector

```diff
@@ -16,7 +16,12 @@ def which(program):
   for path in [""] + os.environ.get("PATH", "").split(os.pathsep):
     exe_file = os.path.join(path, program)
     if os.path.isfile(exe_file) and os.access(exe_file, os.X_OK):
-      return exe_file
+      # in case of pyenv we might have `python3` command available, but it's just a stub
+      # https://github.com/emscripten-core/emscripten/issues/8792
+      p = subprocess.Popen([exe_file, "--version"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
I'm not sure this is enough - when I check locally, `pyenv --version` works (prints the `pyenv 1.2.12`, and exits with `0`). Perhaps we can check for the output starting with `Python `? But maybe there's something more robust.

Reply:
Oh, I guess it would actually error if pyenv didn't find python3 on my system?

Reply:
In the `python_selector` we always check found python commands, not `pyenv` and good part is that python stubs created `pyenv` return 137 exit code in the problematic case. 

I'm more afraid of string parsing, as this might be more false positive results.
Just in case checked some major flavours of python: 
```bash
#cpython 
➜  python --version 
Python 3.7.0

#pypy3
➜ pypy3 --version
Python 3.5.3 (11af55503d5c, May 23 2019, 09:37:40)
[PyPy 7.0.0 with GCC 9.1.1 20190503 (Red Hat 9.1.1-1)]

#stackless
➜ python --version           
Python 3.5.4

#anaconda 
➜ python --version
Python 3.7.0
(anaconda3-5.3.1) 

#miniconda 
➜ python --version
Python 3.6.3 :: Anaconda, Inc.
(miniconda3-4.3.30) 
``` 

As you've mentioned Ubuntu, I did a quick test of calling python3 command on a fresh installation of 19.04: 
```bash
trzeci@ubuntu1904:~$ python

Command 'python' not found, but can be installed with:

sudo apt install python3         # version 3.7.3-1, or
sudo apt install python          # version 2.7.16-1
sudo apt install python-minimal  # version 2.7.16-1

You also have python3 installed, you can run 'python3' instead.

trzeci@ubuntu1904:~$ echo $?
127
trzeci@ubuntu1904:~$ which python
trzeci@ubuntu1904:~$ 

```
Hence it looks like that's system feature, not a stub command

Reply:
Thanks, yeah, I agree. lgtm!

Reply:
> Oh, I guess it would actually error if pyenv didn't find python3 on my system?

It's already guarded by: `if os.path.isfile(exe_file) and os.access(exe_file, os.X_OK):` hence it shouldn't error if `python3` command isn't available. 

Thanks for merging! 
// Cheers